pipeline/lib/metric.ml

```diff
@@ -0,0 +1,134 @@
+module Github = Current_github
+module BenchmarksData = Map.Make (String)
+
+let parse_benchmark_data ~data =
+  Array.fold_left
+    (fun acc datum ->
+      match datum with
+      | [| _commit; benchmark_name; test_name; metrics |] ->
+          let key = Fmt.str "%s__%s" benchmark_name test_name in
+          let parsed_metrics = Yojson.Safe.from_string metrics in
+          BenchmarksData.add key parsed_metrics acc
+      | _ -> acc)
+    BenchmarksData.empty data
+
+let sum_float_list xs = List.fold_left (fun acc x -> acc +. x) 0. xs
+let calc_diff v cv = Float.abs (v -. cv) /. cv *. 100.
+
+let find_avg vs =
+  vs
+  |> List.filter_map (function
+       | key, value when key = "avg" -> Some value
+       | _ -> None)
+  |> function
+  | [ avg ] -> avg
+  | _ -> 0.
+
+let change_threshold = 0.01
```

Comment:
Is this how much change we are okay with? :) Perhaps along with the other config, we can ask projects for the thresholds for each benchmark they are okay with (just a passing thought, i don't know how annoying it will be 😬 )